src/OpenLoco/src/World/Station.cpp

```diff
@@ -873,6 +880,186 @@ namespace OpenLoco
+                    if (roadStationObj->hasFlags(RoadStationFlags::passenger))
+                    {
+                        acceptedCargoTypes |= 1U << roadStationObj->cargoType;
+                    }
+                    else if (roadStationObj->hasFlags(RoadStationFlags::freight))
+                    {
+                        // Exclude passengers from this station type
+                        acceptedCargoTypes |= 0xFFFFFFFFU & ~(1U << roadStationObj->cargoType);
+                    }
+                    else
+                    {
+                        acceptedCargoTypes = 0xFFFFFFFFU;
+                    }
+                    break;
+                }
+                case StationType::airport:
+                    station->flags |= StationFlags::transportModeAir;
+                    acceptedCargoTypes = 0xFFFFFFFFU;
+                    break;
+                case StationType::docks:
+                    station->flags |= StationFlags::transportModeWater;
+                    acceptedCargoTypes = 0xFFFFFFFFU;
+                    break;
+            }
+        }
+
+        for (auto cargoType = 0U; cargoType < ObjectManager::getMaxObjects(ObjectType::cargo); ++cargoType)
+        {
+            auto* cargoObj = ObjectManager::get<CargoObject>(cargoType);
+            if (cargoObj == nullptr)
+            {
+                continue;
+            }
+            station->cargoStats[cargoType].flags &= ~StationCargoStatsFlags::flag1;
+            if (acceptedCargoTypes & (1U << cargoType))
+            {
+                station->cargoStats[cargoType].flags |= StationCargoStatsFlags::flag1;
+            }
+        }
+    }
+
+    // 0x0048F321
+    void addTileToStation(const StationId stationId, const World::Pos3& pos, uint8_t rotation)
+    {
+        auto* station = StationManager::get(stationId);
+        station->stationTiles[station->stationTileSize] = pos;
+        station->stationTiles[station->stationTileSize].z &= ~0x3;
+        station->stationTiles[station->stationTileSize].z |= (rotation & 0x3);
+        station->stationTileSize++;
+
+        CargoSearchState cargoSearchState;
+        const auto acceptedCargos = station->calcAcceptedCargo(cargoSearchState);
+
+        if (cargoSearchState.cargoFilterHasBeenRecalculated() && (station->flags & StationFlags::flag_5) != StationFlags::none)
+        {
+            station->flags &= ~StationFlags::flag_5;
+            auto* town = TownManager::get(station->town);
+            town->numStations++;
+            Ui::WindowManager::invalidate(WindowType::town, enumValue(station->town));
+
+            for (auto& station2 : StationManager::stations())
+            {
+                if (station2.owner == station->owner)
+                {
+                    continue;
+                }
+                // THIS LOOKS WRONG WHY ISN'T IT station2.flags (vanilla mistake?)
+                if ((station->flags & StationFlags::flag_5) != StationFlags::none)
```

Comment:
Must be, right? Though I wonder what the impact of the change would be. Any ideas what flag 5 represents, and how often it is used?

Reply:
I'm pretty sure it's a flag to indicate a station hasn't finished it's creation process yet. I don't know how you would end up with multiple stations without it set though as it's set during the last stage of creation.

Reply:
Hmm, that's odd, I thought station creation was always instantaneous. Perhaps an AI thing?

Reply:
is this something you'd like to address in this PR?

Reply:
I think we will have to look at it after all the other station creation functions are done. Its hard to tell at the moment with only the train station. (Although i'm pretty confident that it is a mistake)